boards/DIM/Inc/App/App_BinarySwitch.h

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include <stdbool.h>
+
+struct BinarySwitch;
+
+struct BinarySwitch *App_BinarySwitch_Create(bool (*is_turned_on)(void));
+void App_BinarySwitch_Destroy(struct BinarySwitch *binary_switch);
+bool App_BinarySwitch_IsTurnedOn(const struct BinarySwitch *binary_switch);
```

Comment:
Hmm yeah I wrote the jdocs and then removed them because I was unsure whether it's worthwhile to add jdocs when it's just rewriting the function names (e.g. Writing `Check if the given switch is turned on` for `_IsTurnedon`). I'll add them back though.